XamarinCommunityToolkit/Effects/Touch/TouchEffect.shared.cs

```diff
@@ -0,0 +1,1208 @@
+			-1,
+			propertyChanged: TryGenerateEffect);
+
+		public static readonly BindableProperty NormalBackgroundImageSourceProperty = BindableProperty.CreateAttached(
+			nameof(NormalBackgroundImageSource),
+			typeof(ImageSource),
+			typeof(TouchEffect),
+			default(ImageSource),
+			propertyChanged: (bindable, oldValue, newValue) =>
+			{
+				GetFrom(bindable)?.ForceUpdateState();
+				TryGenerateEffect(bindable, oldValue, newValue);
+			});
+
+		public static readonly BindableProperty HoveredBackgroundImageSourceProperty = BindableProperty.CreateAttached(
+			nameof(HoveredBackgroundImageSource),
+			typeof(ImageSource),
+			typeof(TouchEffect),
+			default(ImageSource),
+			propertyChanged: (bindable, oldValue, newValue) =>
+			{
+				GetFrom(bindable)?.ForceUpdateState();
+				TryGenerateEffect(bindable, oldValue, newValue);
+			});
+
+		public static readonly BindableProperty PressedBackgroundImageSourceProperty = BindableProperty.CreateAttached(
+			nameof(PressedBackgroundImageSource),
+			typeof(ImageSource),
+			typeof(TouchEffect),
+			default(ImageSource),
+			propertyChanged: (bindable, oldValue, newValue) =>
+			{
+				GetFrom(bindable)?.ForceUpdateState();
+				TryGenerateEffect(bindable, oldValue, newValue);
+			});
+
+		public static readonly BindableProperty NormalBackgroundImageAspectProperty = BindableProperty.CreateAttached(
+			nameof(NormalBackgroundImageAspect),
+			typeof(Aspect),
+			typeof(TouchEffect),
+			default(Aspect),
+			propertyChanged: (bindable, oldValue, newValue) =>
+			{
+				GetFrom(bindable)?.ForceUpdateState();
+				TryGenerateEffect(bindable, oldValue, newValue);
+			});
+
+		public static readonly BindableProperty HoveredBackgroundImageAspectProperty = BindableProperty.CreateAttached(
+			nameof(HoveredBackgroundImageAspect),
+			typeof(Aspect),
+			typeof(TouchEffect),
+			default(Aspect),
+			propertyChanged: (bindable, oldValue, newValue) =>
+			{
+				GetFrom(bindable)?.ForceUpdateState();
+				TryGenerateEffect(bindable, oldValue, newValue);
+			});
+
+		public static readonly BindableProperty PressedBackgroundImageAspectProperty = BindableProperty.CreateAttached(
+			nameof(PressedBackgroundImageAspect),
+			typeof(Aspect),
+			typeof(TouchEffect),
+			default(Aspect),
+			propertyChanged: (bindable, oldValue, newValue) =>
+			{
+				GetFrom(bindable)?.ForceUpdateState();
+				TryGenerateEffect(bindable, oldValue, newValue);
+			});
+
+		public static readonly BindableProperty ShouldSetImageOnAnimationEndProperty = BindableProperty.CreateAttached(
+			nameof(ShouldSetImageOnAnimationEnd),
+			typeof(bool),
+			typeof(TouchEffect),
+			default(bool),
+			propertyChanged: TryGenerateEffect);
+
+		readonly GestureManager gestureManager = new GestureManager();
+
+		VisualElement control;
+
+		public TouchEffect()
+			: base(EffectIds.TouchEffect)
+		{
+#if __ANDROID__
+			if (System.DateTime.Now.Ticks < 0)
+				_ = new Xamarin.CommunityToolkit.Android.Effects.PlatformTouchEffect();
+#elif __IOS__
```

Comment:
You can add `macOS` here as well.

Reply:
What's the preprocessor directive for macOS and Tizen and UWP?